tools/analyze_functions.py

```diff
@@ -0,0 +1,134 @@
+import ast
```

Comment:
james: add a docstr here to better describe the intent of this module and how/why to run it.